File: appweb/src/types/game.ts
export type GameStatus = 'WaitingForPlayers' | 'InProgress' | 'Completed';
export type PieceColorApi = 'Black' | 'White';

export interface GameResponse {
  id: string;
  playerBlackId: string | null;
  playerBlackUsername: string | null;
  playerBlackAvatarUrl: string | null;
  playerWhiteId: string | null;
  playerWhiteUsername: string | null;
  playerWhiteAvatarUrl: string | null;
  winnerId: string | null;
  status: GameStatus;
  boardState: string;
  currentTurn: PieceColorApi;
  createdAt: string;
  updatedAt: string;
}

File: api/Models/Game.cs
using api.Models.Enums;

namespace api.Models;

public class Game
{
    public Guid Id { get; set; }
    public Guid? PlayerBlackId { get; set; }
    public Guid? PlayerWhiteId { get; set; }
    public Guid? WinnerId { get; set; }
    public GameStatus Status { get; set; }
    public string BoardState { get; set; } = string.Empty;
    public PieceColor CurrentTurn { get; set; }
    public long BetAmountSats { get; set; }
    public bool BetSettled { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public Player? PlayerBlack { get; set; }
    public Player? PlayerWhite { get; set; }
    public ICollection<Move> Moves { get; set; } = [];
}
